### PR TITLE
completed assesment

### PR DESCRIPTION
**Pull request by arnoldc94**
# Submitting Week 1 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [ ✅] Coding Challenges
- [ ✅] Coding Concepts
- [ ✅] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -16,41 +24,106 @@ const fruit2 = "banana"
...
 // Pseudo code:
+// create a function named tempEvaluation
+// create the paramater temp to pass in the temperatures value to the function
+// create an if conditional statement that evaluates whether the variable temp is above the boiling point of 212. 
+// if the temp  above 212 return the <temp> is above boiling point.
+// create an else if conditional statement that evaluates whether the variable temp is equal to 212 .
+// if the temp variable is equal to 212 return the <temp> is at boiling point
+// create an else statement that evaluates whether the variable temp is less than  212 .
+// if the temp variable is below 212 return the <temp> is below boiling point
+//invoke the function in console.log below the function and pass in the the paramater
```
Comment (elysemontano) [inline code-challenges.js:50]: Excellent pseudocode!!

File: code-challenges.js [hunk 2]
```diff
@@ -16,41 +24,106 @@ const fruit2 = "banana"
...
+    if(temp > 212){
+        return temp + " is above boiling point"
+    }else if(temp === 212 ){
+        return temp + " is at boiling point"
+    }else{
+        return temp + " is below boiling point"
+    }
+}
+
+console.log(tempEvaluation(temp3))
```
Comment (elysemontano) [inline code-challenges.js:67]: Looks great!

File: code-challenges.js [hunk 3]
```diff
@@ -16,41 +24,106 @@ const fruit2 = "banana"
...
+// Pseudo code: // create a variable that will store the new array combined array.
+// inside the new array write the first array
+// add the concat method and the second array that you want to add to the end of the first array.
+// log the new array and use the length method to display the new length of the array.
 
 const myNumbers1 = [3, 7, 0, 36, -9]
 const myNumbers2 = [8, -7, 42, 9, 13]
 
+let newArray = myNumbers1.concat(myNumbers2)
+console.log(newArray.length)
```
Comment (elysemontano) [inline code-challenges.js:81]: Wonderful!

File: code-challenges.js [hunk 4]
```diff
@@ -16,41 +24,106 @@ const fruit2 = "banana"
...
 
-
+for (i = 0; i < myArray.length; i++){
+    if( myArray[i] % 2 === 0){
+        console.log("even")
+    }else{
+        console.log("odd")
+    }
+}
+console.log(myArray)
```
Comment (elysemontano) [inline code-challenges.js:118]: Code looks great!  For a stretch, what are some edge cases you can test out?

File: code-challenges.js [hunk 5]
```diff
@@ -59,3 +132,15 @@ const number2 = 100
...
+ const subtraction = (num1, num2) =>{
+     if(num1 >= num2){
+         return num1 - num2
+     }else{
+         return num2 - num1
+     }
+ }
+
+ console.log(subtraction(number1, number2))
+ console.log(subtraction(number3, number4))
```
Comment (elysemontano) [inline code-challenges.js:145]: Excellent work in this section!  Code is well written and your pseudocode is outstanding!

File: coding-concepts.js [hunk 6]
```diff
@@ -9,51 +9,51 @@
...
 // --------------------2) What will this log?
 
 const greeting = "Hello World!"
-// console.log(greeting[4])
+//console.log(greeting[4])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: o
+// b) Verify and explain: The answer o is correct. Each character has an index starting with 0, so by accessing index 4 with bracket notation then the result is o. 0 = H, 1 = e, 2 = l, 3 = l, 4 = 0.
```
Comment (elysemontano) [inline coding-concepts.js:31]: Really thorough answer!

File: coding-concepts.js [hunk 7]
```diff
@@ -9,51 +9,51 @@
...
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+//console.log(weekendDays.toUpperCase())
+// a) Your answer: error
+// b) Verify and explain: It is error. The toUpperCase method would be used in this way for strings. In order to cappitalize the contents of the array you would need to iterate over the array using a method or a loop to access each element in the array. The map() method would iterate over each element in the array calling a function on each element in this case capitalizing each one.
```
Comment (elysemontano) [inline coding-concepts.js:50]: Nice!!  This one often stumps a lot of people.  Also like that you are exploring concepts not yet covered in class like .map().

File: interview-questions.md [hunk 8]
```diff
@@ -7,62 +7,63 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Pair programming is a useful tool in the development process. One team member is the driver while the other is the navigator. The driver is responsible for typing out the code (hands on the keyboard). The navigator is responsible for telling the driver what to type and using another set of eyes to see syntax errors, solving problems that happen while figuring out a problem and looking for more efficient ideas to write the code.
+
+  Researched answer: This is an agile development technique where two people work together on code. One person is the navigator who guides the development process.Then the driver is at the keyboard. This can be done remotely or in person and helps detect errors. The team decide on well defined task at a time as they work. The team rotates roles which helps keep developers alert and engaged. Two programmers with matched skills can also use the unstructured method. It is unstructured and hard to organize remotely. Their is the ping pong style which allow one to write a test and the next to pass the test.
```
Comment (elysemontano) [inline interview-questions.md:52]: Outstanding answers in this section!!  Keep up the excellent work!

**Review by elysemontano — APPROVED** [review]
Way to start of these assessments Corey!  This work is excellent!!!  Your logic and pseudocode is great and you are very thorough in your explanations and interview answers.  See if you can stretch yourself next week by exploring something new that perhaps wasn't covered in class.  Keep up the great work!!
